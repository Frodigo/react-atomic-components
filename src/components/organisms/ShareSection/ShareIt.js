import React from 'react';
import PropTypes from 'prop-types';
import {
    EmailShareButton,
    FacebookShareButton,
    FacebookMessengerShareButton,
    LinkedinShareButton,
    TelegramShareButton,
    TwitterShareButton,
    WhatsappShareButton
} from 'react-share';

import Heading from '../../atoms/Text/Heading';
// import FacebookIcon from '../../../public/assets/icons/social/facebook.svg';
// import TwitterIcon from '../../../public/assets/icons/social/twitter.svg';
// import LinkedinIcon from '../../../public/assets/icons/social/linkedin.svg';
// import EmailIcon from '../../../public/assets/icons/social/send-mail.svg';
// import MessengerIcon from '../../../public/assets/icons/social/messenger.svg';
// import TelegramIcon from '../../../public/assets/icons/social/telegram.svg';
// import WhatsappIcon from '../../../public/assets/icons/social/whatsapp.svg';

import classes from './ShareSection.module.css';

const ShareIt = (props) => {
    const { heading, url, ...rest } = props;

    const urlToDisplay = url.url ? url.url : url;

    return (
        <div className={classes.shareSection} {...rest}>
            <Heading type="h4">{heading}</Heading>
            {/*<ul className={classes.shareLinks}>*/}
            {/*    <li className={classes.facebook}>*/}
            {/*        <FacebookShareButton url={urlToDisplay}>*/}
            {/*            <FacebookIcon />*/}
            {/*        </FacebookShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.twitter}>*/}
            {/*        <TwitterShareButton url={urlToDisplay}>*/}
            {/*            <TwitterIcon />*/}
            {/*        </TwitterShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.linkedin}>*/}
            {/*        <LinkedinShareButton url={urlToDisplay}>*/}
            {/*            <LinkedinIcon />*/}
            {/*        </LinkedinShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.mail}>*/}
            {/*        <EmailShareButton url={urlToDisplay}>*/}
            {/*            <EmailIcon />*/}
            {/*        </EmailShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.messenger}>*/}
            {/*        <FacebookMessengerShareButton url={urlToDisplay}>*/}
            {/*            <MessengerIcon />*/}
            {/*        </FacebookMessengerShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.telegram}>*/}
            {/*        <TelegramShareButton url={urlToDisplay}>*/}
            {/*            <TelegramIcon />*/}
            {/*        </TelegramShareButton>*/}
            {/*    </li>*/}
            {/*    <li className={classes.whatsapp}>*/}
            {/*        <WhatsappShareButton url={urlToDisplay}>*/}
            {/*            <WhatsappIcon />*/}
            {/*        </WhatsappShareButton>*/}
            {/*    </li>*/}
            {/*</ul>*/}
            <p>
                Each share supports and motivates me to create new content Thank
                you!
            </p>
        </div>
    );
};

ShareIt.propTypes = {
    heading: PropTypes.string,
    url: PropTypes.object.isRequired
};

ShareIt.defaultProps = {
    heading: 'Did you like it? Share!'
};

export default ShareIt;
