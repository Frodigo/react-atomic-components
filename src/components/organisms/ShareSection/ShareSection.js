import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';
// import FacebookIcon from '../../../public/assets/icons/social/facebook.svg';
// import TwitterIcon from '../../../public/assets/icons/social/twitter.svg';
// import LinkedinIcon from '../../../public/assets/icons/social/linkedin.svg';
// import GithubIcon from '../../../public/assets/icons/social/github.svg';

import classes from './ShareSection.module.css';

const ShareSection = (props) => {
    const { heading, ...rest } = props;

    return (
        <div className={classes.shareSection} {...rest}>
            <Heading type="h4">{heading}</Heading>

        </div>
    );
};

ShareSection.propTypes = {
    heading: PropTypes.string
};

ShareSection.defaultProps = {
    heading: 'Did you like it? Share!'
};

export default ShareSection;
