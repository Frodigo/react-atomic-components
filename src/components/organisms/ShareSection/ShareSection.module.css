.shareSection {
    border-top: solid 2px var(--color-primary);
    padding: var(--spacing-xxl) var(--spacing-md);
    text-align: center;
}

.shareLinks {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin-top: 30px;
}

.shareLinks li {
    margin: 7px;
}

.shareLinks button {
    outline: none;
}

.shareLinks li::before {
    display: none;
}

.shareLink {
    align-items: center;
    display: flex;
    justify-content: center;
    height: 50px;
    width: 50px;
}

.facebook {
    composes: shareLink;
    background-color: var(--color-alternate);
}

.twitter {
    composes: shareLink;
    background-color: var(--color-primary);
}

.linkedin {
    composes: shareLink;
    background-color: var(--color-tiertary);
}

.mail {
    composes: shareLink;
    background-color: var(--color-secondary);
}

.messenger {
    composes: shareLink;
    background-color: var(--color-blue);
}

.telegram {
    composes: shareLink;
    background-color: var(--color-red);
}

.whatsapp {
    composes: shareLink;
    background-color: var(--color-green);
}

@media (min-width: 1300px) {
    .shareLink {
        transform: scale(1);
        transition: transform .3s ease-in-out;
    }

    .shareLink:hover,
    .shareLink:focus {
        transform: scale(1.2);
    }
}
