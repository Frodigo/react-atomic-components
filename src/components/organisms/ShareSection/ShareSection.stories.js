import React from 'react';

import { default as ShareSectionComponent } from './ShareSection';

export default {
    title: 'Organisms',
    component: ShareSectionComponent
};

const Template = (args) => <ShareSectionComponent {...args} />;

export const ShareSection = Template.bind({});
