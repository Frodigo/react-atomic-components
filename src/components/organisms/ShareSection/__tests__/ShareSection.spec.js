import React from 'react';
import { render } from '@testing-library/react';

import ShareSection from '../ShareSection';

describe('ShareSection should be rendered', () => {
    test('ShareSection renders correctly', () => {
        const { container } = render(<ShareSection />);

        expect(container).toMatchSnapshot();
    });
});
