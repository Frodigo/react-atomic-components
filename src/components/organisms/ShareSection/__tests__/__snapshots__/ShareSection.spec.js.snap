// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareSection should be rendered ShareSection renders correctly 1`] = `
<div>
  <div
    class="shareSection"
  >
    <h4
      class=""
    >
      Did you like it? Share!
    </h4>
  </div>
</div>
`;
